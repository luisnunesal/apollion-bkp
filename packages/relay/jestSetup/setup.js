const { JSDOM } = require('jsdom');
const server = require('./testServer');

module.exports = async () => {
  const httpServer = await server.listen(5000, () => {
    console.log(`\n test server running at: \n${JSON.stringify(httpServer.address(), null, 2)}`);
  });

  Object.assign(global, { httpServer });
};
