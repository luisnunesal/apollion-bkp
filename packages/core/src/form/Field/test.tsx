import 'jest-styled-components';

import React from 'react';
import userEvent from '@testing-library/user-event';

import { Field } from '.';

import { render, screen } from '../../__tests__/setup';

describe('Field', () => {
  const EmptyComponent = () => <></>;

  // ---- LABEL ---- //
  it('Should render with label', () => {
    render(<Field component={EmptyComponent} label="Input" name="Input" />);
    expect(screen.queryByLabelText('Input')).toBeDefined();
  });

  it('Should render with hideLabel={true}', () => {
    render(<Field component={EmptyComponent} label="Input" name="Input" hideLabel />);
    expect(screen.queryByLabelText('Input')).toBeNull();
  });

  it('Should render optionalText', () => {
    render(<Field component={EmptyComponent} optionalText="optional" />);

    expect(screen.queryByText('optional')).toBeDefined();
  });

  // ---- VARIANTS ---- //
  it('Snapshot should match variant="default"', () => {
    render(<Field component={EmptyComponent} name="Input" />);
    expect(screen).toMatchSnapshot();
  });

  it('Snapshot should match variant="error"', () => {
    render(<Field component={EmptyComponent} name="Input" error />);
    expect(screen).toMatchSnapshot();
  });

  it('Snapshot should match variant="success"', () => {
    render(<Field component={EmptyComponent} name="Input" success />);
    expect(screen).toMatchSnapshot();
  });

  // ---- HINTS ---- //
  it('Snapshot should match hintText="hint"', () => {
    render(<Field component={EmptyComponent} name="Input" hintText="hint" />);
    expect(screen.getAllByText('hint')).toBeDefined();
  });

  it('Snapshot should match hintSuccessText="success"', () => {
    render(<Field component={EmptyComponent} name="Input" success hintSuccessText="success" />);
    expect(screen.getAllByText('success')).toBeDefined();
  });

  it('Snapshot should match hinterrorText="error"', () => {
    render(<Field component={EmptyComponent} name="Input" error hintErrorText="error" />);
    expect(screen.getAllByText('error')).toBeDefined();
  });

  // ---- onChange ---- //
  it('Snapshot call onChange / setFieldValue', () => {
    const onChangeMockFn = jest.fn();
    const InputComponent = (props: any) => <input id={props.id} name={props.name} onChange={props.onChange} />;

    render(
      <Field
        component={InputComponent}
        name="Input"
        label="Input"
        hintText="hint"
        setFieldValue={onChangeMockFn}
        // inputProps={{ onChange: onChangeMockFn }}
      />,
    );

    const input = screen.getByLabelText('Input') as HTMLInputElement;
    userEvent.type(input, 'changed');

    expect(input.value).toBe('changed');
    expect(onChangeMockFn).toHaveBeenCalledTimes(7);
  });
});
