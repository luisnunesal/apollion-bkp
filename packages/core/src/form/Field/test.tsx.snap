// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field Snapshot should match variant="default" 1`] = `
.c2 {
  position: relative;
  text-rendering: optimizeLegibility;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #4F7F90;
  font-size: 0.9rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0;
  margin: 0;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 > * {
  margin: 0.125rem 0 !important;
}

.c0 > *:first-child {
  margin-top: 0 !important;
}

.c0 > *:last-child {
  margin-bottom: 0 !important;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  visibility: hidden;
}

.c3 {
  font-size: 0.75rem;
  line-height: 1.2rem;
  font-weight: 700;
  color: #19586E;
}

.c4 {
  font-size: 0.6875rem;
  line-height: 1.2rem;
  color: #19586E;
  visibility: hidden;
  min-height: 1.25rem;
}

<div
  class="sc-AxjAm c0"
>
  <div
    class="sc-AxjAm c1"
  >
    <label
      class="c2 c3"
      for="Input"
    />
  </div>
  <label
    class="c2 c4"
  />
</div>
`;

exports[`Field Snapshot should match variant="error" 1`] = `
.c2 {
  position: relative;
  text-rendering: optimizeLegibility;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #4F7F90;
  font-size: 0.9rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0;
  margin: 0;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 > * {
  margin: 0.125rem 0 !important;
}

.c0 > *:first-child {
  margin-top: 0 !important;
}

.c0 > *:last-child {
  margin-bottom: 0 !important;
}

.c0 .c3,
.c0 .sc-AxgMl {
  color: #5A1007;
}

.c0 .c5 {
  color: #E12712;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  visibility: hidden;
}

.c4 {
  font-size: 0.75rem;
  line-height: 1.2rem;
  font-weight: 700;
  color: #19586E;
}

.c6 {
  font-size: 0.6875rem;
  line-height: 1.2rem;
  color: #19586E;
  visibility: visible;
  min-height: 1.25rem;
}

<div
  class="sc-AxjAm c0"
>
  <div
    class="sc-AxjAm c1"
  >
    <label
      class="c2 c3 c4"
      for="Input"
    />
  </div>
  <label
    class="c2 c5 c6"
  >
    Something went wrong
  </label>
</div>
`;

exports[`Field Snapshot should match variant="success" 1`] = `
.c2 {
  position: relative;
  text-rendering: optimizeLegibility;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #4F7F90;
  font-size: 0.9rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0;
  margin: 0;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 > * {
  margin: 0.125rem 0 !important;
}

.c0 > *:first-child {
  margin-top: 0 !important;
}

.c0 > *:last-child {
  margin-bottom: 0 !important;
}

.c0 .c3,
.c0 .sc-AxgMl {
  color: #124829;
}

.c0 .c5 {
  color: #2CB567;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  visibility: hidden;
}

.c4 {
  font-size: 0.75rem;
  line-height: 1.2rem;
  font-weight: 700;
  color: #19586E;
}

.c6 {
  font-size: 0.6875rem;
  line-height: 1.2rem;
  color: #19586E;
  visibility: visible;
  min-height: 1.25rem;
}

<div
  class="sc-AxjAm c0"
>
  <div
    class="sc-AxjAm c1"
  >
    <label
      class="c2 c3 c4"
      for="Input"
    />
  </div>
  <label
    class="c2 c5 c6"
  >
    Everything is alright
  </label>
</div>
`;
